cmake/app/boilerplate.cmake

```diff
@@ -526,9 +526,9 @@ set(PROJECT_SOURCE_DIR ${ZEPHYR_BASE})
 
 set(KERNEL_NAME ${CONFIG_KERNEL_BIN_NAME})
 
-set(KERNEL_ELF_NAME   ${KERNEL_NAME}.elf)
-set(KERNEL_BIN_NAME   ${KERNEL_NAME}.bin)
-set(KERNEL_HEX_NAME   ${KERNEL_NAME}.hex)
+set(KERNEL_ELF_NAME   ${KERNEL_NAME}.elf CACHE PATH "Kernel ELF file name")
+set(KERNEL_BIN_NAME   ${KERNEL_NAME}.bin CACHE PATH "Kernel bin file name")
+set(KERNEL_HEX_NAME   ${KERNEL_NAME}.hex CACHE PATH "Kernel HEX file name")
```

Comment:
Are we sure we want to put those into the cache ?
The CMake cache entries are considered user settable:
https://cmake.org/cmake/help/v3.17/command/set.html#set-cache-entry
> Sets the given cache <variable> (cache entry). Since cache entries are meant to provide user-settable values this does not overwrite existing cache entries by default. Use the FORCE option to overwrite existing entries.

but in this particular case, the name is based on the Kconfig setting `CONFIG_KERNEL_BIN_NAME`, so if user changes the config name later, the cache is not updated.
(Or if the user knows CMake and thinks he can modify the cache, other things might break.)

Can we get this name directly from the Kconfig file in python ?